### PR TITLE
add video_insights stream and some other tweaks

### PR DESCRIPTION
**Pull request by yujoy**

### CODE REVIEW

File: tap_facebook_pages/streams.py [hunk 1]
```diff
@@ -534,6 +600,182 @@ def parse_response(self, response: requests.Response) -> Iterable[dict]:
                             yield values
 
 
+# class RecentPostEngagementInsightsStream(RecentInsightsStream):
```
Comment (prratek) [inline tap_facebook_pages/streams.py:603]: What are all the commented out streams? Is this the work that was in `develop`?

File: tap_facebook_pages/streams.py [hunk 2]
```diff
@@ -117,9 +117,75 @@ def get_url_params(
         params = super().get_url_params(context, next_page_token)
         params["access_token"] = self.page_access_tokens[context["page_id"]]
         params["fields"] = ",".join(self.schema["properties"].keys())
+        if self.get_starting_timestamp(context):
```
[the remark this answers is not in the document]
Comment (yujoy) [reply tap_facebook_pages/streams.py:120]: I think this was in response to [this slack msg](https://voxmedia.slack.com/archives/C037EAZ1YSC/p1671478516718859?thread_ts=1671474155.366169&cid=C037EAZ1YSC), but it didn't really do anything. Should we keep it?

File: tap_facebook_pages/streams.py [hunk 3]
```diff
@@ -117,9 +117,75 @@ def get_url_params(
...
+            params["since"] = (
+                pendulum.instance(self.get_starting_timestamp(context))
+                .to_date_string()
+            )
+        else:
+            # `since` date isn't included in the range so subtract 1 day
+            params["since"] = (
+                pendulum.parse(self.config["start_date"])
+                .subtract(days=1)
```
[the remark this answers is not in the document]
Comment (yujoy) [reply tap_facebook_pages/streams.py:129]: Yeah we can, but only if we decide to keep it from the comment above ^

File: tap_facebook_pages/streams.py [hunk 4]
```diff
@@ -117,9 +117,75 @@ def get_url_params(
...
+                and date(created_time) >= date_sub(current_date, interval @insights_lookback_months month)
+
+            union distinct
+
+            select distinct id, created_time
+            from `g9-data-warehouse-prod.facebook_videos.all`
+            where
+                `from`.id = @page_id
+                and date(created_time) >= date_sub(current_date, interval @insights_lookback_months month)
+        """
```
[the remark this answers is not in the document]
Comment (yujoy) [reply tap_facebook_pages/streams.py:165]: Mmm good idea, but yeah I'll leave it for now so we can get to QA-ing quicker

File: tap_facebook_pages/streams.py (hunk 1 above)
Comment (yujoy) [reply tap_facebook_pages/streams.py:603]: Yeah

File: tap_facebook_pages/streams.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (prratek) [reply tap_facebook_pages/streams.py:120]: ah okay maybe comment it out or remove it for now and we can figure out how we want to handle pagination for both videos and posts after we finish insights

**Review by prratek — APPROVED** [review]
looks good after we comment out the pagination changes

**Pull request merged**